src/utils/bytes.js

```diff
@@ -24,12 +24,14 @@ import {dev} from '../log';
  */
 export function utf8Decode(bytes) {
   if (TextDecoder) {
-    return Promise.resolve(new TextDecoder('utf-8').decode(bytes));
+    return new Promise(resolve => {
```

Comment:
`TextDecoder#decode` doesn't return a promise, and it throws synchronously on errors, which is not the intended behavior for `utf8Decode`. Also, promise resolvers (the function passed to the constructor) as executed sync, so it's not deferred. But, the resolver is wrapped in a try-catch in the internal Promise code, so if it throws sync it will result in the new promise being rejected.